src/state/browserHistory/index.ts

```diff
@@ -15,8 +15,8 @@ interface BrowserHistoryStore {
 
 const MAX_RECENT_SIZE = 1000;
 
-export const useBrowserHistoryStore = createRainbowStore<BrowserHistoryStore>(
-  set => ({
+export const useBrowserHistoryStore = createRainbowStore<BrowserHistoryStore & { hasVisited: (url: string) => boolean }>(
```

Comment:
if you decide to keep this in the store, you should add the type directly inside the `BrowserHistoryStore` type instead of tacking it on here